README.md

```diff
@@ -3,4 +3,356 @@
+```
+
+Then assign the permissions to the `Functions` profile with
+
+```sh
+sfdx force:user:permset:assign -n Functions
+```
+
+## Create and Run a Python Function Locally
+
+### Generate the Python Function
+
+From the SFDX project root, run:
+
+```sh
+sf generate function \
+  --language python \
+  --function-name hellopython
+```
+
+The remaining commands will be executed within the newly generated project folder. Change to that folder by running:
+
+```sh
+cd functions/hellopython
+```
+
+You should also create a `.gitignore` file in the function directory with the following contents:
+```gitignore
+venv
+__pycache__
+```
+
+The following command will do this for you:
+```sh
+echo "venv\n__pycache__" > .gitignore
+```
+
+### Create the Python [Virtual Environment](https://docs.python.org/3/library/venv.html#creating-virtual-environments) & Install Dependencies
+
+To install the dependencies required by the Python Function locally, we first need to create a "Virtual Environment" (venv) 
+which we can install packages into without affecting your system Python installation.  This can be done by running:
+```sh
+python3 -m venv venv
+```
+
+Next, the virtual environment needs to be activated.  
+
+On a **macOS / Linux system** you can activate the virtual environment with
+
+```sh
+source venv/bin/activate
+```
+
+On a **Microsoft Windows system** you can activate the virtual environment with
+```sh
+.\venv\Scripts\activate
+```
+
+> For help with setting up a virtual environment, see the [Python documentation](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#creating-a-virtual-environment).
+
+Finally, the dependencies can be installed into the newly created environment:
+
+```sh
+pip3 install -r requirements.txt
+```
+
+
+This will ensure your function has all dependencies it requires installed before you run it.  If you forgot to do this 
+before starting the Function locally you’ll receive an error reminding you to perform this step.
+
+### Run the Python Function Locally
+
+```sh
+sf run function start
+```
+
+This will start the function running locally on http://localhost:8080.  Messages logged by the running function will appear here.
+
+### Invoke the Running Python Function
+
+After starting the function:
+
+* open a new command line terminal 
+* navigate to the `hellopython` directory 
+* invoke the function by sending it a payload
+
+```sh
+sf run function --function-url http://localhost:8080 --payload '{}'
+```
+
+## Deploy the Python Function
+
+The following commands need to be executed from the **root of your SFDX project** so change into that directory now.
+
+### Commit your changes to Git
+
+All code changes made to a function will need to be staged and committed before you can deploy.
+
+```sh
+git add . 
+git commit -m "Trying out python functions"
+```
+
+Once everything is committed to git, run:
+
+```sh
+sf deploy functions --connected-org PythonScratch
```

Comment:
This command failed for me, with:

```
$ sf deploy functions --connected-org PythonScratch
Pushing changes to functions... failed
    Error: Request failed with status code 404
```

It looks like the guide is missing the "create the compute env" step?
https://developer.salesforce.com/docs/platform/functions/guide/deploy.html#create-the-compute-environment

After I ran this:

```
$ sf env create compute --connected-org PythonScratch --alias PythonEnv
Creating compute environment for org ID 00DJO00000004Rp2AI... done
New compute environment created with ID pythonf-00djo00000004rp2ai-670
Connecting environments... done
Your compute environment with local alias PythonEnv is ready.
```

...then the `sf deploy functions --connected-org PythonScratch` command worked.

Reply:
Oh I missed this step in the guide (I was switching between the rendered view and raw markdown to leave review comments, so made it harder to keep track).

The 404 is pretty bad UX from the Salesforce CLI though - I'll file an issue to get that improved.

Reply:
Short term I've added a note to the docs in #34:
https://github.com/heroku/sf-functions-python/blob/a7541831525ae1850ae2c8511d2b0c8f94597a53/README.md#L289

Longer term, I've filed GUS-W-12244992 for fixing `plugin-functions` to handle the 404 case gracefully - which it seems like it was supposed to be doing already, but isn't for some reason:
https://github.com/salesforcecli/plugin-functions/blob/6d8fb47c6927aa89d31eacf91069bae767ab5578/src/commands/deploy/functions.ts#L85-L96